### PR TITLE
Source Notion: add 'synced block' property and date format

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What
Small fixes for Source Notion Silver Certification.
1. Added properties for `synced_block` Block type. 
2. Added date-time format to various properties in Comments and shared File schemas.

## How
Synced blocks can consist of "original synced blocks" or "duplicated sync blocks":
  - In the former case, the block contains a Null `synced_from` property, and may contain a `children` array of Blocks objects.
  - In the latter case, the `synced_from` property contains a `type` of `block_id` and a `block_id` key; these blocks have no `children` array.
  
[Docs link](https://developers.notion.com/reference/block#synced-block)
  
## User Impact
Backwards compatibility tests were failing due to the addition of the "date-time" format for the specified fields. However, my understanding is these changes should **not** constitute a breaking change, as this PR contains only additions of missing properties and formats. I'd like to confirm that this is correct and we can safely merge without necessitating a breaking change migration.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`2573552136`](https://github.com/airbytehq/airbyte/pull/31742/commits/257355213658b88a8e5be698ca8f6dba1fb05e93)) - ❌

⏲️  Total pipeline duration: 04mn03s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Build source-notion docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                   | ✅       |
| Acceptance tests                                             | ✅       |
| Check our base image is used                                 | ✅       |
| Code format checks                                           | ✅       |
| Validate metadata for source-notion                          | ✅       |
| Connector version semver check                               | ✅       |
| Connector version increment check                            | ✅       |
| QA checks                                                    | ❌       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_notion-schema-fix/1698092056/257355213658b88a8e5be698ca8f6dba1fb05e93/source-notion/2.0.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:257355213658b88a8e5be698ca8f6dba1fb05e93)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`6dd4f54f46`](https://github.com/airbytehq/airbyte/pull/31742/commits/6dd4f54f4680632422adfeaba3623eb7f1d872fd)) - ✅

⏲️  Total pipeline duration: 03mn34s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Build source-notion docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                   | ✅       |
| Acceptance tests                                             | ✅       |
| Check our base image is used                                 | ✅       |
| Code format checks                                           | ✅       |
| Validate metadata for source-notion                          | ✅       |
| Connector version semver check                               | ✅       |
| Connector version increment check                            | ✅       |
| QA checks                                                    | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_notion-schema-fix/1698092616/6dd4f54f4680632422adfeaba3623eb7f1d872fd/source-notion/2.0.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:6dd4f54f4680632422adfeaba3623eb7f1d872fd)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`085beeead2`](https://github.com/airbytehq/airbyte/pull/31742/commits/085beeead23949bd3175060770ca612850abdd4b)) - ✅

⏲️  Total pipeline duration: 03mn53s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Build source-notion docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                   | ✅       |
| Acceptance tests                                             | ✅       |
| Check our base image is used                                 | ✅       |
| Code format checks                                           | ✅       |
| Validate metadata for source-notion                          | ✅       |
| Connector version semver check                               | ✅       |
| Connector version increment check                            | ✅       |
| QA checks                                                    | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_notion-schema-fix/1698093654/085beeead23949bd3175060770ca612850abdd4b/source-notion/2.0.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:085beeead23949bd3175060770ca612850abdd4b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

File: docs/integrations/sources/notion.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ The connector is restricted by Notion [request limits](https://developers.notion
 
 | Version | Date       | Pull Request                                             | Subject                                                                      |
 | :------ | :--------- | :------------------------------------------------------- | :--------------------------------------------------------------------------- |
+| 2.0.5   | 2023-10-23 | [31742](https://github.com/airbytehq/airbyte/pull/31742) | Add 'synced_block' property to Blocks schema |
```
Comment (davydov-d) [inline docs/integrations/sources/notion.md:115]: nit: please format this table :pray:

**Review by davydov-d — APPROVED** [review]

**Comment by airbyte-oss-build-runner** [comment]
##  source-notion test report (commit [`50104de08e`](https://github.com/airbytehq/airbyte/pull/31742/commits/50104de08e921e4a5e05b78ba14f1111ee7383dc)) - ✅

⏲️  Total pipeline duration: 03mn55s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Build source-notion docker image for platform(s) linux/amd64 | ✅       |
| Unit tests                                                   | ✅       |
| Acceptance tests                                             | ✅       |
| Check our base image is used                                 | ✅       |
| Code format checks                                           | ✅       |
| Validate metadata for source-notion                          | ✅       |
| Connector version semver check                               | ✅       |
| Connector version increment check                            | ✅       |
| QA checks                                                    | ✅       |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/christo_notion-schema-fix/1698156583/50104de08e921e4a5e05b78ba14f1111ee7383dc/source-notion/2.0.5/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:50104de08e921e4a5e05b78ba14f1111ee7383dc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-notion test
```

**Pull request merged**